website/src/app/blog/using-tauri/readme.mdx

```diff
@@ -0,0 +1,270 @@
+- Win32 API - Verbose, complicated, documentation is confusing, not portable to
+  Linux
+- native-windows-gui - Not popular enough / not maintained
+- WinForms / WPF - Binding to C# assumed to be difficult
+- Electron - Huge downloads, quality of Rust bindings unknown
+
+# What is good about Tauri?
+
+## Tauri is easy to learn and use
+
+Using Tauri for the first time, I got the Windows Client to MVP in about 2
+months, and the Linux Client to MVP in another 4 months. Linux took longer
+because its security model required us to split the Client into a privileged
+tunnel service and un-privileged GUI.
+
+(TODO: Screenshots of the Windows and Linux MVPs)
+
+Software that is easy for beginners to use ends up attracting beginners and
+turning them into experienced users and contributors. The Rust language
+deliberately took advantage of this, shipping the beginner-friendly Cargo
+package manager from Rust 1.0 and shipping Clippy a few years later in 2018.
+
+Tauri does the same thing, shipping the `tauri-cli` tool for setting up
+boilerplate projects, and package bundlers for Windows `.msi` installers and
+Debian `.deb` packages that Just Work, at least for simple programs.
+
+## It's Rust and TypeScript
+
+Firezone already uses Rust for the data plane (the Gateway, the Relay, and
+`connlib`) and TypeScript for the website and admin portal. I don't know much
+web design, so using TypeScript allowed me to make a rough prototype of the GUI
+and then let other team members polish and style it to match the website
+branding.
+
+JavaScript inside the web view can call into `async` Rust functions (TODO: Code
+example), and from there we can use `tokio`'s mpsc channels to send events to a
+central controller task. (TODO: Code)
+
+# What snags did we hit?
+
+Overall, I like Tauri a lot. This section is longer because there are lots of
+little details. "It's good" is shorter than "We had trouble because..."
+
+## The Tauri directory layout is odd
+
+The root of a Tauri project, such as
+[`/rust/gui-client` in the Firezone repo](https://github.com/firezone/firezone/tree/2a1187bd9c4d12960ef0000df1b6bd9bb361bf10/rust/gui-client),
+has two subdirectories: `src` and `src-tauri`. The entire Rust project is
+contained in `src-tauri`, and the `src` directory contains files for the web
+view. `Cargo.toml` and `tauri.conf.json` are in `src-tauri`. I find this
+confusing. I think Tauri's primary purpose is to wrap up web apps into native
+apps, so the Rust code is sequestered in `src-tauri`, and the project is
+supposed to be an NPM project that incidentally contains a Rust wrapper.
+
+I think of the Firezone Client as a Rust GUI program that incidentally uses
+HTML+JS+CSS for the GUI, so this layout is backwards to me. I would have
+expected a top-level `/Cargo.toml`, with the Rust under `/src`, and the web
+files under `/src-web` or similar. Maybe there's a reason for this, but I'm not
+aware of it.
+
+## Linux doesn't like running GUIs with root permissions
```

Comment:
Interesting section! I think it's a good candidate for much more detail on the approaches we used for Windows and Linux specifically, and what other approaches we considered (I remember that design document you had drafted that had a few different options like single binary but run in different modes...)